imap_processing/lo/l1a/lo_l1a.py

```diff
@@ -347,3 +356,65 @@ def add_dataset_attrs(
             dataset[var].attrs.pop("DEPEND_0")
 
     return dataset
+
+
+def instrument_state_vector(datasets_by_apid_derived: dict) -> xr.Dataset:
+    """
+    Combine relevant datasets to create instrument state vector.
+
+    This is from section 9.1.1 of the IMAP-Lo Algorithm Document. We
+    combine the NHK, SHK, and PCC datasets to create a single data
+    product containing the instrument state vector. These are not
+    all aligned in time, so we simply merge them and fill NaNs where
+    the times are not aligned across data products.
+
+    Parameters
+    ----------
+    datasets_by_apid_derived : dict
+        Dictionary of datasets keyed by APID with derived values.
+
+    Returns
+    -------
+    xr.Dataset
+        Combined dataset containing the instrument state vector.
+    """
+    datasets_to_merge = [
+        ds
+        for apid, ds in datasets_by_apid_derived.items()
+        if apid in [LoAPID.ILO_APP_NHK, LoAPID.ILO_APP_SHK, LoAPID.ILO_DIAG_PCC]
+    ]
+    combined_ds = xr.merge(datasets_to_merge, compat="override")
+    # Only keep the desired keys for the state vector
+    state_vector_keys = [
```

Comment:
Are the an_c_thr, tof3_thr, tof2_thr, tof1_thr, tof0_thr, ifb_hot_spot_t, coarse_pot_pri, and fine_pot_pri no longer needed in this?

Reply:
!! Thank you for actually looking at this, I just didn't continue to the second page. Updating now...